powerscale/provider/network_pool_resource.go

```diff
@@ -0,0 +1,588 @@
+			"description": schema.StringAttribute{
+				Description:         "A description of the pool.",
+				MarkdownDescription: "A description of the pool.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"groupnet": schema.StringAttribute{
+				Description:         "Name of the groupnet this pool belongs to.",
+				MarkdownDescription: "Name of the groupnet this pool belongs to.",
+				Required:            true,
+			},
+			"id": schema.StringAttribute{
+				Description:         "Unique Pool ID.",
+				MarkdownDescription: "Unique Pool ID.",
+				Computed:            true,
+			},
+			"ifaces": schema.ListNestedAttribute{
+				Description:         "List of interface members in this pool.",
+				MarkdownDescription: "List of interface members in this pool.",
+				Optional:            true,
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"iface": schema.StringAttribute{
+							Description:         "A string that defines an interface name.",
+							MarkdownDescription: "A string that defines an interface name.",
+							Optional:            true,
+							Computed:            true,
+						},
+						"lnn": schema.Int64Attribute{
+							Description:         "Logical Node Number (LNN) of a node.",
+							MarkdownDescription: "Logical Node Number (LNN) of a node.",
+							Optional:            true,
+							Computed:            true,
+						},
+					},
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "The name of the pool. It must be unique throughout the given subnet.It's a required field with POST method.",
+				MarkdownDescription: "The name of the pool. It must be unique throughout the given subnet.It's a required field with POST method.",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtMost(255),
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[0-9a-zA-Z_-]*$`),
+						"must contain only alphanumeric characters and _-",
+					),
+				},
+			},
+			"nfsv3_rroce_only": schema.BoolAttribute{
+				Description:         "Indicates that pool contains only RDMA RRoCE capable interfaces.",
+				MarkdownDescription: "Indicates that pool contains only RDMA RRoCE capable interfaces.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"ranges": schema.ListNestedAttribute{
+				Description:         "List of IP address ranges in this pool.",
+				MarkdownDescription: "List of IP address ranges in this pool.",
+				Optional:            true,
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"high": schema.StringAttribute{
+							Description:         "High IP",
+							MarkdownDescription: "High IP",
+							Optional:            true,
+							Computed:            true,
+							Validators: []validator.String{
+								stringvalidator.RegexMatches(
```

Comment:
for these regex, are there test cases for testing invalid string?

Reply:
These regex are specified by powerscale json file. I locally did some tests to validate their function.

Reply:
@RayLiu7  yes, the regex is provided by the json file, but in my side, Some long regex string actually are not well matched.

Reply:
Yes, I understand. Actually I've tried some invalid combinations I can come up with already. The reason why I didn't put them in ATs is that it's handled by Terraform and will not influenced by future logic changes we make in CRUD functions. Thus don't need to be run every time in pipeline run.

Reply:
As we discussed, it's better to remove local regex check and leave it to api, because the regex in json file and api are not always corresponding. We may accidentally block some inputs that are actually acceptable for api.